src/executor/execute.rs

```diff
@@ -70,89 +71,77 @@ pub async fn execute<T: 'static + Debug, U: Store<T> + StoreMut<T> + AlterTable>
-        }
-        Prepared::Delete(keys) => {
-            let num_rows = keys.len();
-
-            storage
-                .delete_data(keys)
-                .await
-                .map(|(storage, _)| (storage, Payload::Delete(num_rows)))
-        }
-        Prepared::Update(table_name, all_column_defs, columns_to_update, rows) => {
-            let all_column_defs = Rc::from(all_column_defs);
-            let validated = validate_rows(
-                &storage,
-                &table_name,
-                ColumnValidation::SpecifiedColumns(Rc::clone(&all_column_defs), columns_to_update),
-                rows.iter().map(|r| &r.1),
-            )
-            .await;
-            try_into!(storage, validated);
-
-            let num_rows = rows.len();
+        Statement::Drop {
+            object_type,
+            names,
+            if_exists,
+            ..
+        } => {
+            if object_type != &ObjectType::Table {
+                return Err((storage, ExecuteError::DropTypeNotSupported.into()));
+            }
 
-            storage
-                .update_data(rows)
+            stream::iter(names.iter().map(Ok))
+                .try_fold(storage, |storage, table_name| async move {
+                    let schema = try_block!(storage, {
+                        let table_name = get_name(table_name)?;
+                        let schema = storage.fetch_schema(table_name).await?;
+
+                        if !if_exists {
+                            schema.ok_or(ExecuteError::TableNotExists)?;
+                        }
+                        Ok(table_name)
+                    });
+                    storage
+                        .delete_schema(schema)
+                        .await
+                        .map(|(storage, _)| storage)
+                })
                 .await
-                .map(|(storage, _)| (storage, Payload::Update(num_rows)))
+                .map(|storage| (storage, Payload::DropTable))
         }
-        Prepared::DropTable {
-            schema_names,
-            if_exists,
-        } => stream::iter(schema_names.into_iter().map(Ok::<&str, (U, Error)>))
-            .try_fold(storage, |storage, table_name| async move {
-                let schema = try_into!(storage, storage.fetch_schema(table_name).await);
-
-                if !if_exists {
-                    try_into!(storage, schema.ok_or(ExecuteError::TableNotExists));
-                }
-
-                storage
-                    .delete_schema(table_name)
-                    .await
-                    .map(|(storage, _)| storage)
-            })
-            .await
-            .map(|storage| (storage, Payload::DropTable)),
-        Prepared::Select { labels, rows } => Ok((storage, Payload::Select { labels, rows })),
 
         #[cfg(feature = "alter-table")]
-        Prepared::AlterTable(table_name, operation) => {
+        Statement::AlterTable { name, operation } => {
```

Comment:
Your decision to keep `try_into!` make sense, it is still useful in `AlterTable`.
However, there are no other use cases except for `AlterTable`. so, Let's move `try_into!` to here.

Then you can remove `#[allow(unused_macros)]`

Reply:
Hmmm it doesn't let me put it there
But I've just realised I could just use the feature item for the macro